### PR TITLE
PLAT-82 record-message serializer

### PR DESCRIPTION
**Pull request by cyraxred**
**- What I did**
* protoc-based code generation that does all necessary work, including:
** no more nullable=false etc
** zero-size types are skipped without nullable (zero-able, not null-able)
** types are registered for deserialization
** code and checks for notation compliance
** generation of head(s)
** etc
* protoc-gen-ins with extra options and logic
* some tricks to change protoc-gen-gogo behavior without fork

### CODE REVIEW

File: ledger-core/v2/reference/global_test.go [hunk 1]
```diff
@@ -42,17 +42,65 @@ func randLocal() Local {
 	return id
 }
 
+func flipHeader(b []byte) []byte {
+	// Encoder uses BigEndian, while AsBytes uses LittleEndian for the header
+	// b[0], b[1], b[2], b[3] = b[3], b[2], b[1], b[0]
+	return b
```
Comment (sexyhamster) [inline ledger-core/v2/reference/global_test.go:48]: something is missing over here or function is not needed anymore

Comment (cyraxred) [reply ledger-core/v2/reference/global_test.go:48]: This is stashed as first I'd like to merge this branch, and then fix the encoding.

File: ledger-core/v2/reference/scope.go [hunk 2]
```diff
@@ -27,47 +31,79 @@ const ( // super-scopes
 )
 
 const ( // super-scopes
-	LifelineSuperScope    Scope = 0x04 * baseScopeLifeline
-	LocalDomainSuperScope Scope = 0x04 * baseScopeLocalDomain
-	GlobalSuperScope      Scope = 0x04 * baseScopeGlobal
+	superScopeLifeline    Scope = 0x04 * baseScopeLifeline
```
Comment (ivanshibitov) [inline ledger-core/v2/reference/scope.go:34]: I would give name to 0x04

Comment (cyraxred) [reply ledger-core/v2/reference/scope.go]: done
593407f8c87af0ebb5d4301e5e2983510f2aad7d

**Pull request merged**